src/devices/bus/nes/mmc3_clones.cpp

```diff
@@ -109,13 +109,33 @@ nes_8237_device::nes_8237_device(const machine_config &mconfig, const char *tag,
 {
 }
 
-nes_sglionk_device::nes_sglionk_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock)
-	: nes_txrom_device(mconfig, NES_SG_LIONK, tag, owner, clock), m_reg(0), m_reg_enabled(0)
+nes_sglionk_device::nes_sglionk_device(const machine_config &mconfig, device_type type, const char *tag, device_t *owner, u32 clock, const u8 *reg_table, const u16 *addr_table)
+	: nes_txrom_device(mconfig, type, tag, owner, clock)
+	, m_mmc3_mode(true)
+	, m_reg_table(reg_table)
+	, m_addr_table(addr_table)
 {
 }
 
-nes_sgboog_device::nes_sgboog_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock)
-	: nes_txrom_device(mconfig, NES_SG_BOOG, tag, owner, clock), m_mode(0)
+nes_sglionk_device::nes_sglionk_device(const machine_config &mconfig, const char *tag, device_t *owner, u32 clock)
+	: nes_sglionk_device(mconfig,
+			     NES_SG_LIONK,
+			     tag,
+			     owner,
+			     clock,
+			     (const u8 []) {0, 3, 1, 5, 6, 7, 2, 4},
+			     (const u16 []) {0xa001, 0xa000, 0x8000, 0xc000, 0x8001, 0xc001, 0xe000, 0xe001})
```

Comment:
Is this actually safe?  Strictly speaking, aren’t these temporaries?  I get the feeling this works by coincidence due to how the compiler has implemented it.

Reply:
Damn, my C++-fu is not strong enough to answer this question. In C++11 a temporary object's lifetime is extended if bound to a const reference, but I'm not sure if the same rule applies portably to pointers in this case.

Reply:
Yeah, but the const reference it’s bound to there would just be the argument to the base class constructor, which only has the lifetime of the base class constructor invocation, so it wouldn’t help you.

Reply:
Well regardless of whether it's kosher, I can't really defend the choices here. I guess I was trying hard to be creative? I'm submitting a more traditional, staid patch.